plugins/modules/postgresql_user.py

```diff
@@ -274,6 +288,22 @@
   community.postgresql.postgresql_user:
     name: monitoring
     priv: 'pg_catalog.pg_stat_database:SELECT'
+
+# Create a user and set a default-configuration that is active when they start a session
+- name: Create a user with config-parameter
+  community.postgresql.postgresql_user:
+    name: appclient
+    password: "secret123"
+    configuration:
+      - work_mem=16MB
```

Comment:
```suggestion
    configuration:
      - work_mem=16MB
```
How about making it a dict instead, it'd look like:
```
    configuration:
      work_mem: 16MB
      purge: True
```

Reply:
I have been thinking about various ways to do this. The dict method has the issue that parameters can include dots (e.g. `pg_audit.log`) which would need to be quoted in YAML which might inconvenience users who forget or don't know that.

I also thought about this, which I find too bulky:
```
configuration:
  - key: work_mem
    value: 16MB
  - key: ...
    value: ...
```
Also I'm not too happy with the way I'm doing it it right now.  
Let me know which one you prefer.

Reply:
I'm personally in favor of that one i posted even if it needs quoting (we can make a short note about this):
```
    configuration:
      work_mem: 16MB
      purge: True
```
I'm not necessarily against of having it as in the PR now though. The dict form looks better to me but it's subjective. Also we usually handle similar cases with dicts here and there.
@hunleyd @betanummeric what's your feeling?

Reply:
Okay, I changed it in that way, but I added a separate parameter to enable purging on an empty dict, otherwise this can get really confusing

Reply:
The
```
configuration:
  - work_mem=16MB
```
style means that we need to implement custom parsing and document that to distinguish key and value. What if the key contains a `=` or some other special chars?
I would prefer the
```
configuration:
  work_mem: 16MB
```
style because it uses standard yaml parsing and also prevents duplicate keys at paring level.

Yes, yaml may have some quirks new users may stumble upon, but if we make something custom it's more to learn for everybody.

Reply:
> Okay, I changed it in that way, but I added a separate parameter to enable purging on an empty dict, otherwise this can get really confusing

did you push this change @toydarian ? i don't see a difference here